include/nuttx/lib/math32.h

```diff
@@ -29,6 +29,35 @@
 
 #include <stdint.h>
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Returns one plus the index of the most significant 1-bit of n,
+ * or if n is zero, returns zero.
+ */
+
+#if UINTPTR_MAX > UINT32_MAX
+#define FLS(n) ((n) & 0xffffffff00000000ull ? 32 + \
```

Comment:
```suggestion
#define FLS(n) ((n) & UINT64_C(0xffffffff00000000) ? 32 + \
```
Improve portability by using the conversion macros from `inttypes.h`.
For 64-bit architectures (e.g. RV64) the `ul` should be sufficient.

Reply:
yeah, but many places will need such update. `ull` should work for RV64 as well

Reply:
ok, thank you.